wlutil/config.py

```diff
@@ -411,6 +419,20 @@ def applyBase(self, baseCfg):
             self.cfg['run'] = getOpt('wlutil-dir') / 'null_run.sh'
             self.cfg['runSpec'] = RunSpec(script=self.cfg['run'])
 
+        # To avoid needlessly recompiling kernels, we check if the child has
+        # the exact same binary-related configuration.
+        self.cfg['use-parent-bin'] = True
+        for opt in ['firmware-src', 'host-init']:
+            if opt in self.cfg and self.cfg.get(opt, None) != baseCfg.get(opt, None):
+                self.cfg['use-parent-bin'] = False
+
+        if 'linux' in self.cfg and 'linux' in baseCfg:
+            for opt in self.cfg['linux'].keys():
+                if self.cfg['linux'].get(opt, None) != baseCfg['linux'].get(opt, None):
```

Comment:
What if both `linux` options have identically named configs/files but relative to different workdirs?  Are the paths fully resolved when comparing them?

Reply:
All paths are converted to absolute early in the config loading process. For the linux stuff this happens at line 285 and 287.